### PR TITLE
[DNM] Use static metrics for storage/txn

### PR DESCRIPTION
**Pull request by breeswish**
Do Not Merge: This PR currently replaces some local metrics with static metrics, which later proved to have negative effects. This kind of change will be reverted.

## What have you changed? (mandatory)

Please refer to #3225

## What are the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

No test is needed. Once it is compiled successfully, it works.

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No.

## Does this PR affect tidb-ansible update? (mandatory)

No.

## Refer to a related PR or issue link (optional)

This PR contains changes from #3226 , so you may want to merge that PR first.

### CODE REVIEW

File: src/storage/txn/scheduler.rs [hunk 1]
```diff
@@ -940,17 +928,14 @@ impl Default for SchedContext {
...
-            command_keyread_duration: KV_COMMAND_KEYREAD_HISTOGRAM_VEC.local(),
             command_gc_skipped_counter: KV_COMMAND_GC_SKIPPED_COUNTER.local(),
             command_gc_empty_range_counter: KV_COMMAND_GC_EMPTY_RANGE_COUNTER.local(),
         }
     }
 }
 
 impl SchedContext {
-    fn add_statistics(&mut self, cmd_tag: &'static str, stat: &Statistics) {
+    fn add_statistics(&mut self, cmd_tag: String, stat: &Statistics) {
```
Comment (siddontang) [inline src/storage/txn/scheduler.rs]: why String here? do we need to allocate a String now?

Comment (breeswish) [reply src/storage/txn/scheduler.rs]: It is just a stack allocation. I will make a PR to keep current `&'static`. It needs to add new methods to the static-metric macro.

Comment (breeswish) [reply src/storage/txn/scheduler.rs]: https://github.com/pingcap/rust-prometheus/pull/183

**Comment by breeswish** [comment]
Mixing local() with static metrics is too ugly. Closing for now. Will be replaced by a better solution.

**Pull request closed without merging**